bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java

```diff
@@ -40,59 +48,75 @@
 
-    private final ItemRegistry itemRegistry;
     private final ProfileCallback callback;
-    private List<Class<? extends State>> acceptedDataTypes;
 
-    private List<StateCondition> conditions = List.of();
+    private final List<StateCondition> conditions;
 
-    private @Nullable State configMismatchState = null;
+    private final @Nullable State configMismatchState;
 
     public StateFilterProfile(ProfileCallback callback, ProfileContext context, ItemRegistry itemRegistry) {
         this.callback = callback;
-        acceptedDataTypes = context.getAcceptedDataTypes();
-        this.itemRegistry = itemRegistry;
 
         StateFilterProfileConfig config = context.getConfiguration().as(StateFilterProfileConfig.class);
         if (config != null) {
-            conditions = parseConditions(config.conditions, config.separator);
-            configMismatchState = parseState(config.mismatchState);
+            conditions = parseConditions(config.conditions, config.separator, itemRegistry);
+            if (conditions.isEmpty()) {
+                logger.warn("No valid conditions defined for StateFilterProfile. Link: {}. Conditions: {}",
+                        callback.getItemChannelLink(), config.conditions);
+            }
+            configMismatchState = parseState(config.mismatchState, context.getAcceptedDataTypes());
+        } else {
+            conditions = List.of();
+            configMismatchState = null;
         }
     }
 
-    private List<StateCondition> parseConditions(@Nullable String config, String separator) {
-        if (config == null) {
-            return List.of();
-        }
-
+    private List<StateCondition> parseConditions(List<String> conditions, String separator, ItemRegistry itemRegistry) {
         List<StateCondition> parsedConditions = new ArrayList<>();
-        try {
-            String[] expressions = config.split(separator);
-            for (String expression : expressions) {
-                String[] parts = expression.trim().split("\s");
-                if (parts.length == 3) {
-                    String itemName = parts[0];
-                    StateCondition.ComparisonType conditionType = StateCondition.ComparisonType
-                            .valueOf(parts[1].toUpperCase(Locale.ROOT));
-                    String value = parts[2];
-                    parsedConditions.add(new StateCondition(itemName, conditionType, value));
-                } else {
-                    logger.warn("Malformed condition expression: '{}'", expression);
-                }
-            }
 
-            return parsedConditions;
-        } catch (IllegalArgumentException e) {
-            logger.warn("Cannot parse condition {}. Expected format ITEM_NAME <EQ|NEQ> STATE_VALUE: '{}'", config,
-                    e.getMessage());
-            return List.of();
-        }
+        String operatorPattern = Stream.of(StateCondition.ComparisonType.values())
```

Comment:
You could make that (and the resulting `Pattern`) a static constant and improve performance.

Reply:
done